### ISSUE TITLE
Find node by attribute value

### ISSUE THREAD

**Issue by senier** [Componolit/SXML#50]
The `Query.Path` operation is convenient for finding nodes by name in the XML tree. Another common requirement is finding a node by the existence of an attribute or by an attribute value. the `Query.Path` could be extended to support constructs like this:

```
/parent/child/grandchild[@nodename=nodevalue]
```

- [x] First, the `Query.Find_Attribute` operation must be extended to support searching for an attribute value (once the attribute name has found). This could be a parameter with default value set to "*" (every attribute value matches).

- [x] Next, the `Query.Find_Sibling` operation is extended with parameters for attribute name and value. Both should have "*" as default value. ~~If the attribute value is non-default, the attribute name must be non-default, too (precondition).~~ (Querying a node with an arbitrary attribute having a specific value may be an application).

- [x] As a last step, `Query.Path` can be extended to parse the `[@nodename=nodevalue]` expression and translate it into a respective `Query.Find_Sibling` query.

**Comment by senier** [comment]
Closed via d756802782c8a06f59263df234dfbb5e0a5f3170

**Comment by senier** [comment]
We need to support attribute matching at every level of the query, e.g.:

```
/gexf/graph/attributes[@class=node]/attribute[@title=kind]
```
